examples/data-objects/inventory-app/webpack.config.cjs

```diff
@@ -20,7 +18,7 @@ module.exports = (env) => {
 				extensionAlias: {
 					".js": [".ts", ".tsx", ".js", ".cjs", ".mjs"],
 				},
-				extensions: [".ts", ".tsx", ".js", ".cjs", ".mjs"],
+				extensions: [".js"],
```

Comment:
As far as I can tell the webpack config docs ( https://webpack.js.org/configuration/resolve/#resolveextensionalias ) don't explain how extensionAlias and extensions relate to each other, but by testing it empirically, it appears we only need to list the output of the aliasing in the list of extensions, so I have removed the unneeded entries.

Reply:
That has also been my understanding. Since we are being explicit with paths we may not need this at all. The default is [.js, .json, .wasm]. So, this makes it more restrictive. The thing I don't know is if we need .cjs / .mjs for anything we depend on that doesn't have an explicit extension. We might only find that out in a package that encounters such a dependency.
Setting `enforceExtension: true` would be good.
The above `extensionAlias` setting has been confirmed incorrect. We currently think the correct general setting to be:
```json
			extensionAlias: {
				".cjs": [".cts", ".cjs"],
				".js": [".ts", ".tsx", ".js"],
				".mjs": [".mts", ".mjs"],
			},
```

Reply:
I can confirm we work without extensions, as well as with random garbage as the extensions list (so its not that default works, but that anything works). I think since we can build with enforceExtension, then `extensions` is never used.

Thus I'll remove `extensions` and add `enforceExtension: true`.

For extensionAlias I'd do as you recommend, but I'll omit explicit cjs support: we don't require it and I suspect its mainly used when mjs/esm is also available which we would prefer to use.

Reply:
Annoyingly, webpack's hot module reloading code fails with enforeceExtensions:

```
ERROR in ../../../node_modules/.pnpm/webpack@5.89.0_webpack-cli@4.10.0/node_modules/webpack/hot/emitter.js 1:19-36
Module not found: Error: Can't resolve 'events' in '/home/craig/Work/FluidFramework/node_modules/.pnpm/webpack@5.89.0_webpack-cli@4.10.0/node_modules/webpack/hot'
The request 'events' failed to resolve only because 'resolve.enforceExtension' was specified.
```

I can confirm we work without extensions. As omitting it gets us a default with wasm support which might be nice some day, and maybe some third party lib will use json too, seems good to use the default there I guess.

For extensionAlias I'd do as you recommend, but I'll omit explicit cjs support: we don't require it and I suspect its mainly used when mjs/esm is also available which we would prefer to use.

Reply:
It was merged but I just thought of one question, do we need `".mts"` as a mapping target? I did not see any instances of `.mts` files being used in our repo. @jason-ha